CMakeLists.txt

```diff
@@ -14,14 +14,46 @@ project(
 include(cmake/project-is-top-level.cmake)
 include(cmake/variables.cmake)
 
+find_package(PkgConfig REQUIRED)
+
+find_package(pcre QUIET)
+
+if (NOT pcre_FOUND)
+    # vcpkg provides pcre cmake config as unofficial-pcre
+    find_package(unofficial-pcre CONFIG REQUIRED)
+    add_library(pcre::libpcre ALIAS unofficial::pcre::pcre)
+endif ()
+
+find_package(pcre2 QUIET)
+
+if (NOT pcre2_FOUND)
+    pkg_check_modules(pcre2 REQUIRED libpcre2-8 IMPORTED_TARGET)
+    add_library(pcre2::pcre2 ALIAS PkgConfig::pcre2)
+endif ()
+
+find_package(readline QUIET)
+
+if (NOT readline_FOUND)
+    pkg_check_modules(readline REQUIRED readline IMPORTED_TARGET)
```

Comment:
same with readline, typically system redline is shared library but vcpkg gets static linking